galgebra/ga.py

```diff
@@ -1263,52 +1272,70 @@ def wedge(self, A, B):
         if self.is_ortho:
-            dot_product_basis_blades = self.dot_product_basis_blades
+            dot_product_basis_blades = lambda x: self.dot_product_basis_blades(x, mode=mode)
         else:
-            dot_product_basis_blades = self.non_orthogonal_dot_product_basis_blades
+            dot_product_basis_blades = lambda x: self.non_orthogonal_dot_product_basis_blades(x, mode=mode)
 
-        if self.dot_mode == '|':  # Hestenes dot product
+        if mode == '|':  # Hestenes dot product
             A = self.remove_scalar_part(A)
             B = self.remove_scalar_part(B)
             return update_and_substitute(A, B, dot_product_basis_blades, self.dot_table_dict)
-        elif self.dot_mode == '<' or self.dot_mode == '>':
+        elif mode == '<' or mode == '>':
+            r"""
+            Let :math:`A = a + A'` and :math:`B = b + B'` where :math:`a` and
+            :math:`b` are the scalar parts of :math:`A` and :math:`B`, and
+            :math:`A'` and :math:`B'` are the remaining parts of :math:`A` and
+            :math:`B`. Then we have:
+
+            .. math::
+
+                (a+A') \rfloor (b+B') &= a(b+B') + A' \rfloor B' \\
+                (a+A') \lfloor (b+B') &= b(a+A') + A' \lfloor B'
+
+            We use these relations to reduce :math:`A \rfloor B` (``A<B``) and 
+            :math:`A \lfloor B` (``A>B``).
+            """
             (a, Ap) = self.split_multivector(A)  # Ap = A'
             (b, Bp) = self.split_multivector(B)  # Bp = B'
-            if self.dot_mode == '<':  # Left contraction
+            if mode == '<':  # Left contraction
                 if Ap != 0 and Bp != 0:  # Neither nc part of A or B is zero
                     prod = update_and_substitute(Ap, Bp, dot_product_basis_blades, self.left_contract_table_dict)
                     return prod + a * B
                 else:  # Ap or Bp is zero
                     return a * B
-            elif self.dot_mode == '>':  # Right contraction
+            elif mode == '>':  # Right contraction
                 if Ap != 0 and Bp != 0: # Neither nc part of A or B is zero
                     prod = update_and_substitute(Ap, Bp, dot_product_basis_blades, self.right_contract_table_dict)
                     return prod + b * A
                 else:  # Ap or Bp is zero
                     return b * A
         else:
-            raise ValueError('"' + str(self.dot_mode) + '" not a legal mode in dot')
+            raise ValueError('"' + str(mode) + '" not a legal mode in dot')
+
+    def hestenes_dot(self, A, B):
+        r""" compute the hestenes dot product, :math:`A \bullet B` """
+        return self._dot(A, B, mode='|')
+
+    def left_contract(self, A, B):
+        r""" compute the left contraction, :math:`A \rfloor B`  """
+        return self._dot(A, B, mode='<')
+
+    def right_contract(self, A, B):
+        r""" compute the right contraction, :math:`A \lfloor B` """
+        return self._dot(A, B, mode='>')
```

Comment:
Names are up for debate here, although it probably doesn't matter since any actual user would use operator overloads on `Mv`

Reply:
Shouldn't they be private then if they're not part of the API?

Reply:
They're part of the public API of ga just like `dot` is, but chances are no one is using that API, and people are using the `Mv` API